types/views/views.go

```diff
@@ -190,6 +210,22 @@ type Slice[T any] struct {
 	ж []T
 }
 
+// MapKey returns a unique key for a slice, based on its address and length.
+// It does not hash the contents of the slice.
+func (v Slice[T]) MapKey() SliceMapKey[T] { return mapKey(v.ж) }
+
+// mapKey returns a unique key for a slice, based on its address and length.
+// It does not hash the contents of the slice.
+func mapKey[T any](x []T) SliceMapKey[T] {
```

Comment:
Seems like we are recreating `unsafe.SliceData()` ?

Reply:
kinda i guess? but that operates on cap, this operates on len.